frontend/src/components/file-explorer/tree-node.tsx

```diff
@@ -84,13 +84,17 @@ function TreeNode({ path, defaultOpen = false }: TreeNodeProps) {
         type={isDirectory ? "button" : "submit"}
         name="file"
         value={path}
+        onClick={handleClick}
         className="flex items-center justify-between w-full px-1"
       >
         <Title
           name={filename}
           type={isDirectory ? "folder" : "file"}
           isOpen={isOpen}
-          onClick={handleClick}
+          onClick={(e) => {
+            e.stopPropagation();
+            handleClick();
+          }}
```

Comment:
I'm on my phone so I can't verify right now, but it should be safe to remove this since we are handling the click event on its parent (and by extension this child)